### PR TITLE
Refactor: Iteration 3 methods and tests

### PR DESCRIPTION
**Pull request by alerrian**
In total 9 methods were refactored to around 3 lines each. Each module was used extensively for methods that were necessary for function and were generic enough that using them outside of stat_tracker would be viable. This PR is specifically to get this work in to dev so that there is a clean base to work from come Friday when we are going to be getting access to the next steps. I expanded the scope of some of the tests for these methods to bring to ensure function was more in line with how the rspec harness is supposed to be functioning, despite it not being completely compliant with the rspec harness due to an unknown error through that testing suite. I now have ensured that each method is returning Strings, as expected by the harness.

### CODE REVIEW

File: lib/modules/gatherable.rb [hunk 1]
```diff
@@ -15,6 +15,20 @@ def goals_by_team
...
 
+  def home_goals_by_team
+    @game_collection.collection.inject(Hash.new(0)) do |scores, game|
+      scores[game[1].home_team_id] += game[1].home_goals.to_i
+      scores
+    end
+  end
+
+  def away_goals_by_team
+    @game_collection.collection.inject(Hash.new(0)) do |scores, game|
```
[the remark this answers is not in the document]
Comment (alerrian) [reply lib/modules/gatherable.rb]: Already discussed from the last PR.

File: test/stat_tracker_test.rb [hunk 2]
```diff
@@ -122,18 +122,47 @@ def test_stat_tracker_gets_count_of_teams
...
+  end
+
+  def test_stat_tracker_can_get_lowest_scoring_home
+    assert_instance_of String, @stat_tracker.lowest_scoring_home_team
+    assert_equal "Utah Royals FC", @stat_tracker.lowest_scoring_home_team
+  end
+
+  def test_stat_tracker_can_get_winningest_team
+    assert_instance_of String, @stat_tracker.winningest_team
+    assert_equal "Reign FC", @stat_tracker.winningest_team
```
[the remark this answers is not in the document]
Comment (alerrian) [reply test/stat_tracker_test.rb]: Whoops. Old habits and all that jazz.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -106,4 +106,32 @@ def worst_defense
...
+  end
+
+  def lowest_scoring_home_team
+    team_id = team_average_goals(home_goals_by_team).min_by{ |_id, avg| avg }[0]
+
+    get_team_name_by_id(team_id)
+  end
+
+  def winningest_team
+
```
[the remark this answers is not in the document]
Comment (alerrian) [reply lib/stat_tracker.rb]: This is a method frame, so the trailing space is necessary to estimate spacing.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -106,4 +106,32 @@ def worst_defense
...
+    get_team_name_by_id(team_id)
+  end
+
+  def lowest_scoring_home_team
+    team_id = team_average_goals(home_goals_by_team).min_by{ |_id, avg| avg }[0]
+
+    get_team_name_by_id(team_id)
+  end
+
+  def winningest_team
```
[the remark this answers is not in the document]
Comment (alerrian) [reply lib/stat_tracker.rb]: I never knew about this and will now keep it in mind.

**Comment by iEv0lv3** [comment]
Just finished reviewing this PR and it is almost ready to merge!

* Functionality works as anticipated
* Methods are on-point in terms of size and complexity
* Use of modules and inheritance is excellent
* I think for the `stat_tracker_test` tests we'll want to make a few of them more assertive to ensure we're getting the expected results.
  * Example: line 125 of `stat_tracker_test.rb`: 
`assert_instance_of String, @stat_tracker.best_offense` should be updated to expect the exact string from the command. As currently written this is similar to using regular `assert`, which gives us a truthy result, but it could really be anything such as a blank string being returned, thus erroneously passing the tests.
* All issues from Hound are related to syntax, and can be quickly updated. I think it would be beneficial to do this now and then merge the PR. My thought process is that if we're going to do a big merge into `master` we should be as close to 'production ready' as possible, which includes syntax. The single-click 'Format Document' from within VS Code using Rubocop should make this process fast and easy.

Overall, outstanding work on the methods! It really reduced the size and complexity of our files. After the formatting changes are pushed into the PR I'll go ahead and merge it into `dev`!

**Review by iEv0lv3 — APPROVED** [review]
Thank you for updating what we discussed, much appreciated. The PR looks great and is now ready to be merged into `dev`.

**Pull request merged**